pages/_app.tsx

```diff
@@ -16,8 +16,33 @@
 
 import '../styles/globals.css'
 import type {AppProps} from 'next/app'
+import { useRouter } from 'next/router';
+import { useState, useEffect } from 'react';
 
 function MyApp({Component, pageProps}: AppProps) {
+    const router = useRouter();
+    const [loading, setLoading] = useState(false);
+    useEffect(() => {
+        const start = () => {
+            setLoading(true);
+        };
+        const end = () => {
+            setLoading(false);
+        };
+        router.events.on("routeChangeStart", start);
+        router.events.on("routeChangeComplete", end);
+        router.events.on("routeChangeError", end);
+        return () => {
+            router.events.off("routeChangeStart", start);
+            router.events.off("routeChangeComplete", end);
+            router.events.off("routeChangeError", end);
+        };
+    }, [router.events]);
```

Comment:
I ever try this trick, but didn't add `[router.events]` part..

Reply:
But from my local env `Loading...` doesn't show.

Reply:
Sigh, it worked briefly, and I now also find that it doesn't seem to work.